Arriba/Arriba/Model/Table.cs

```diff
@@ -354,13 +354,27 @@ public void AddColumnsFromBlock(DataBlock values)
         /// <param name="values">Set of Columns and values to add or update</param>
         public void AddOrUpdate(DataBlock values)
         {
+            AddOrUpdateOptions options = new Model.AddOrUpdateOptions();
+            options.AddMissingColumns = ShouldAddColumnsDynamically;
```

Comment:
Given the new options class, it seems like we can get rid of the ShouldAddColumnsDynamically property.